src/tests/ftest/util/storage_utils.py

```diff
@@ -0,0 +1,629 @@
+
+  SPDX-License-Identifier: BSD-2-Clause-Patent
+"""
+from collections import OrderedDict
+from functools import partial
+import itertools
+from operator import is_not
+import os
+import re
+
+from ClusterShell.NodeSet import NodeSet
+
+from run_utils import run_remote
+
+
+def find_pci_address(value):
+    """Find PCI addresses in the specified string.
+
+    Args:
+        value (str): string to search for PCI addresses
+
+    Returns:
+        list: a list of all the PCI addresses found in the string
+
+    """
+    digit = '0-9a-fA-F'
+    pattern = rf'[{digit}]{{4,5}}:[{digit}]{{2}}:[{digit}]{{2}}\.[{digit}]'
+    return re.findall(pattern, str(value))
+
+
+class StorageException(Exception):
+    """Exception for the StorageInfo class."""
+
+
+class StorageDevice():
+    """Information about a storage device."""
+
+    def __init__(self, address, storage_class, device, numa_node):
+        """Initialize a StorageDevice object.
+
+        Args:
+            address (str): the address of the device
+            storage_class (str): the device class description
+            device (str): the device description
+            numa_node (str): the NUMA node number
+        """
+        self.address = address
+        self.storage_class = storage_class
+        self.device = device
+        self.numa_node = numa_node
+        self.managed_devices = 0
+
+    def __str__(self):
+        """Convert this StorageDevice into a string.
+
+        Returns:
+            str: the string version of the parameter's value
+
+        """
+        return ' - '.join([str(self.address), self.description, str(self.numa_node)])
+
+    def __repr__(self):
+        """Convert this StorageDevice into a string representation.
+
+        Returns:
+            str: raw string representation of the parameter's value
+
+        """
+        return self.__str__()
+
+    def __eq__(self, other):
+        """Determine if this StorageDevice is equal to another StorageDevice.
+
+        Args:
+            other (StorageDevice): the other object to compare
+
+        Returns:
+            bool: whether or not this StorageDevice is equal to another StorageDevice
+
+        """
+        return str(self) == str(other)
+
+    def __lt__(self, other):
+        """Determine if this StorageDevice is less than the other StorageDevice.
+
+        Used to sort devices by their addresses, but with more performant devices listed first.
+
+        Args:
+            other (StorageDevice): the other object to compare
+
+        Returns:
+            bool: True if only the this StorageDevice is more performant, False if the other
+                StorageDevice is more performant, or True if this StorageDevice address is less
+                than the other StorageDevice.
+
+        """
+        if 'optane' in self.device.lower() and 'optane' not in other.device.lower():
+            # This device is more performant than the other device
+            return True
```

Comment:
This is `__lt__`, so to be clear, `dev1 < dev2` means `dev1 is more performant than dev2`? I.e. "less is better"?

Reply:
Yes, so when we sort the list of drives we get the more performant disks before the less performant disks in the list.